keel-web/src/test/kotlin/com/netflix/spinnaker/keel/apidocs/ApiDocTests.kt

```diff
@@ -351,5 +339,21 @@ class ApiDocTests : JUnit5Minutests {
         .booleanValue()
         .isTrue()
     }
+
+    listOf(
+      VersionedArtifactProvider::artifactName.name,
+      VersionedArtifactProvider::artifactType.name,
+      VersionedArtifactProvider::artifactVersion.name,
+    ).forEach { propertyName ->
+      test("ClusterSpec does not contain transient property $propertyName used for image resolution") {
+        at("/\$defs/ClusterSpec/properties/$propertyName")
+          .isA<MissingNode>()
+      }
+
+      test("TitusClusterSpec does not contain transient property $propertyName used for image resolution") {
+        at("/\$defs/TitusClusterSpec/properties/$propertyName")
+          .isA<MissingNode>()
+      }
+    }
```

Comment:
It wasn't clear to me that these properties were being excluded from the JSON schema so I added a test to make sure they are